src/test/java/com/mabl/integration/jenkins/MablRestApiClientTest.java

```diff
@@ -341,4 +344,16 @@ private void registerPostCreateRetryMappings(
                         .withBody(MablTestConstants.CREATE_DEPLOYMENT_EVENT_RESULT_JSON))
         );
     }
+
+    private Secret mockSecret(String value) {
```

Comment:
Would `Secret.fromString()` work?

Reply:
Not sure you need reflective access here, and I don't think we're returning a mock object, but rather an actual Secret object. Per @DonMcNamara 's comment, can we just mock the method of interest?

Alternatively, `Mockito` may be able to just make a mock for us on it's own, e.g. `mock(Secret.class)`.

Reply:
Let me try this!

Reply:
@DonMcNamara Secret.fromString() does not work as it requires Jenkins to be initialized and running, which is not available during test run.

Reply:
@twistedpair - unfortunately, mocking does not seem to work as the hudson.util.Secret class is *final*. Mockito will fail. I don't like resorting to reflection, but it works.